IpmiFeaturePkg/GenericIpmi/Common/GenericIpmi.h

```diff
@@ -32,4 +44,118 @@
   SM_IPMI_BMC_SIGNATURE \
   )
 
-#endif // _IPMI_BMC_H_
+//
+// Dxe Ipmi instance data
+//
+typedef struct {
+  UINTN             Signature;
+  UINT64            IpmiTimeoutPeriod;
+  UINT8             SlaveAddress;
+  UINT8             TempData[MAX_TEMP_DATA];
+  BMC_STATUS        BmcStatus;
+  UINT64            ErrorStatus;
+  UINT8             SoftErrorCount;
+  IPMI_TRANSPORT    IpmiTransport;
+} IPMI_BMC_INSTANCE_DATA;
+
+#pragma pack(1)
+
+//
+// Structure of IPMI Command buffer
+//
+#define IPMI_COMMAND_HEADER_SIZE  2
+
+typedef struct {
+  UINT8    Lun         : 2;
+  UINT8    NetFunction : 6;
+  UINT8    Command;
+  UINT8    CommandData[MAX_TEMP_DATA - IPMI_COMMAND_HEADER_SIZE];
+} IPMI_COMMAND;
+
+//
+// Structure of IPMI Command response buffer
+//
+#define IPMI_RESPONSE_HEADER_SIZE  3
+
+typedef struct {
+  UINT8    Lun         : 2;
+  UINT8    NetFunction : 6;
+  UINT8    Command;
+  UINT8    CompletionCode;
+  UINT8    ResponseData[MAX_TEMP_DATA - IPMI_RESPONSE_HEADER_SIZE];
+} IPMI_RESPONSE;
+
+//
+// Structure to communicate BMC state from PEI to DXE.
+//
+
+typedef struct _IPMI_BMC_HOB {
+  BMC_STATUS    BmcStatus;
+} IPMI_BMC_HOB;
+
+#pragma pack()
+
+/**
+  Initializes the IPMI state for the BMC. This includes performs platform
+  specific logic, getting the device ID, and checking self-test results.
+
+  @param[in,out]  IpmiInstance    The IPMI instance being initialized.

```

Comment:
Is it really in/out?

Reply:
The BmcStatus field is changes to signify the state of the init, I think that qualifies as 'out'